api/src/task/notifications/BaseNotification.py

```diff
@@ -0,0 +1,51 @@
+import logging
+from abc import ABC, abstractmethod
+from typing import Optional
+from uuid import UUID
+
+from sqlalchemy import select
+
+import src.adapters.db as db
+from src.adapters.search import SearchClient
+from src.db.models.user_models import User, UserSavedOpportunity, UserSavedSearch
+from src.task.notifications.constants import EmailData
+
+logger = logging.getLogger(__name__)
+
+
+class BaseNotification(ABC):
```

Comment:
These classes should all be derived from the Task class, just so it comes with the automatic timing/metric stuff. We should have these be used like the transform job where we just have a config to turn tasks on/off. It's perfectly fine to have an ECS task call multiple "Task" classes (even if that ends up being a tad confusing to follow).

You could even make it so this case class has a few abstract methods as you do here, but its `run_task` method calls each of those in order.